### PR TITLE
Enable mtests under Windows with MSVC

### PR DESCRIPTION
**Pull request by AntonioBL**
Resolves: https://musescore.org/en/node/303707
This PR enables the compilation of mtests under Windows with msvc and it also adjusts some of them to pass also under Windows, so that Windows developers using msvc can locally run mtests.
A big thank you to Dmitri Ovodok (dmitrio95) for the part of the code correcting the workspace tests.
Steps for running the mtests can be found in the post in the issue tracker.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [N/A] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by AntonioBL** [comment]
To be redundant, here are the steps, which I verified only by command line compiling:
- Compile MuseScore for Windows (I used Debug version, and I used the msvc_build.bat script) and install it (e.g. to its default location MuseScore\msvc.install_x64
- cd MuseScore\msvc.build_x64\mtest
- cmake --build .
- cmake --build . --target INSTALL
- add the binary dir to the PATH, e.g. PATH=%PATH%;C:\MuseScore\msvc.install_x64\bin
- add the folder where diff.exe for Windows is located (for example msys2/usr/bin if msys2 is installed) to the PATH
- run the tests with ctest

File: mtest/zerberus/comments/CMakeLists.txt [hunk 1]
```diff
@@ -18,4 +18,8 @@ include_directories(
       ${SNDFILE_INCDIR}
       )
 
-target_link_libraries(tst_sfzcomments zerberus synthesizer audiofile ${SNDFILE_LIB} testutils)
+if (NOT MSVC)
+      target_link_libraries(tst_sfzcomments zerberus synthesizer audiofile ${SNDFILE_LIB} testutils)
+else (NOT MSVC)
+      target_link_libraries(tst_sfzcomments zerberus synthesizer audiofile sndfiledll testutils)
+endif (NOT MSVC)
```
Comment (mattmcclinch) [inline mtest/zerberus/comments/CMakeLists.txt:25]: I would rewrite this (and the 6 others like it) as follows:
```
if (MSVC OR MINGW)
      target_link_libraries(tst_sfzcomments zerberus synthesizer audiofile sndfiledll testutils)
else (MSVC OR MINGW)
      target_link_libraries(tst_sfzcomments zerberus synthesizer audiofile ${SNDFILE_LIB} testutils)
endif (MSVC OR MINGW)
```
See https://github.com/musescore/MuseScore/blob/a5c92e2040dae24a09e4cd8bc2a50e61ba2cd579/mscore/CMakeLists.txt#L525-L529 for reference.

Comment (AntonioBL) [reply mtest/zerberus/comments/CMakeLists.txt]: Thank you. Done.

**Comment by Marr11317** [comment]
Thank you! It works like a charm!
For QtCreator users: here's what I did.
image

**Comment by Marr11317** [comment]
> To be redundant, here are the steps, which I verified only by command line compiling:

About being redundant, could you perhaps add the build steps under mtest/README.md?

**Comment by AntonioBL** [comment]
Added instructions to README.md (please let me know if they are clear enough) and rebased on top of current master.

**Comment by Jojo-Schmitz** [comment]
Travis CI is not happy, see https://travis-ci.org/github/musescore/MuseScore/jobs/676875018#L4232-L4240, a build error for the mtests (not a plain test error)

File: mtest/importmidi/tst_importmidi.cpp [hunk 2]
```diff
@@ -20,20 +20,20 @@
 #include "libmscore/chord.h"
 #include "libmscore/note.h"
 #include "libmscore/keysig.h"
-#include "mscore/exportmidi.h"
+#include "importexport/midiexport/exportmidi.h"
```
Comment (mattmcclinch) [inline mtest/importmidi/tst_importmidi.cpp:23]: This file cannot be found. Instead, try
```
#include "audio/exports/exportmidi.h"
```

Comment (AntonioBL) [reply mtest/importmidi/tst_importmidi.cpp]: Thank you.
In these days, MuseScore files are continuosly changing place in the source code.

**Pull request merged**